File: package/umt_python/src/date/from_unix.py
from datetime import datetime, timezone

from .unix_time_unit import UnixTimeUnit


def from_unix(value: int | float, unit: UnixTimeUnit = "s") -> datetime:
    """
    Create a datetime from a Unix timestamp.

    Default unit is seconds. The result is a naive local datetime whose
    epoch offset matches JavaScript Date.getTime().

    Args:
        value: Timestamp value
        unit: "s" for seconds or "ms" for milliseconds

    Returns:
        datetime for the given epoch offset

    Example:
        >>> from_unix(0).timestamp()
        0.0
    """
    milliseconds = value * 1000 if unit == "s" else value
    aware = datetime.fromtimestamp(milliseconds / 1000, tz=timezone.utc)
    return aware.astimezone().replace(tzinfo=None)

File: package/main/src/Date/fromUnix.ts
import type { UnixTimeUnit } from "./unixTimeUnit";

/**
 * Creates a Date from a Unix timestamp.
 * Default unit is seconds.
 *
 * @param {number} value - Timestamp value
 * @param {UnixTimeUnit} [unit="s"] - Seconds or milliseconds
 * @returns {Date} Date for the given epoch offset
 * @example
 * fromUnix(0).getTime(); // 0
 * fromUnix(1_700_000_000_123, "ms").getTime(); // 1700000000123
 */
export const fromUnix = (value: number, unit: UnixTimeUnit = "s"): Date => {
  switch (unit) {
    case "s": {
      return new Date(value * 1000);
    }
    case "ms": {
      return new Date(value);
    }
    default: {
      const _exhaustive: never = unit;
      return _exhaustive;
    }
  }
};
